src/test/components/MenuButtons.test.js

```diff
@@ -14,56 +14,210 @@ import {
 } from '../../actions/receive-profile';
 import { createGeckoProfileWithJsTimings } from '../fixtures/profiles/gecko-profile';
 import { processProfile } from '../../profile-logic/process-profile';
+import { TextEncoder } from 'util';
+import * as ProfileViewSelectors from '../../selectors/profile';
+
+// Mocking SymbolStoreDB
+import { uploadBinaryProfileData } from '../../profile-logic/profile-store';
+jest.mock('../../profile-logic/profile-store');
+
+// Mocking sha1
+import sha1 from '../../utils/sha1';
+jest.mock('../../utils/sha1');
+
+// Mocking compress
+jest.mock('../../utils/gz');
+
+// Mocking shortenUrl
+import shortenUrl from '../../utils/shorten-url';
+jest.mock('../../utils/shorten-url');
 
 describe('app/MenuButtons', function() {
   function setup(profile) {
     const store = storeWithProfile(profile);
-    const { container } = render(
+    const renderResult = render(
       <Provider store={store}>
         <MenuButtons />
       </Provider>
     );
 
+    const { getByTestId, getByValue } = renderResult;
+    const getShareButton = () => getByValue('Share...');
+    const getInnerShareButton = () => getByValue('Share');
+    const getShareWithUrlsButton = () => getByValue('Share with URLs');
+    const getInnerShareWithUrlsButton = (): null | HTMLElement => {
+      const shareWithUrlsButton = getShareWithUrlsButton();
+      if (
+        shareWithUrlsButton.parentElement !== undefined &&
+        shareWithUrlsButton.parentElement !== null &&
+        shareWithUrlsButton.parentElement.nextElementSibling !== undefined &&
+        shareWithUrlsButton.parentElement.nextElementSibling !== null
+      ) {
+        return shareWithUrlsButton.parentElement.nextElementSibling.getElementsByClassName(
+          'arrowPanelOkButton'
+        )[0];
+      }
+      return null;
+    };
+
+    const getMenuButtonsContainer = () =>
+      getByTestId('menuButtonsCompositeButtonContainer');
+    const getPermalinkButton = () => getByValue('Permalink');
+
     return {
       store,
-      container,
+      ...renderResult,
+      getShareButton,
+      getInnerShareButton,
+      getShareWithUrlsButton,
+      getInnerShareWithUrlsButton,
+      getMenuButtonsContainer,
+      getPermalinkButton,
     };
   }
 
-  // profile.meta.networkURLsRemoved flag is set to false as a default.
-  it('renders the MenuButtons buttons', () => {
-    const { store, container } = setup();
-    store.dispatch(startSymbolicating());
+  describe('share and save buttons', function() {
+    // Mock hash
+    const hash = 'c5e53f9ab6aecef926d4be68c84f2de550e2ac2f';
 
-    // MenuButtons is rendering a fragment with several children. We need to
-    // check all children to assess that the component renders properly.
-    expect(Array.from(container.children)).toMatchSnapshot();
+    beforeAll(function() {
+      (window: any).TextEncoder = TextEncoder;
```

Comment:
I think it would be good to check that the TextEncoder doesn't already exist before we mock one out like this. This way if it ever starts existing as a global for whatever reason, we won't fail.

```
beforeAll(function() {
  if ((window: any).TextEncoder) {
    throw new Error("A TextEncoder was already on the window object.");
  }
  (window: any).TextEncoder = TextEncoder;
});
```